src/content/docs/apm/agents/go-agent/installation/install-automation-new-relic-go.mdx

```diff
@@ -0,0 +1,97 @@
+---
+title: 'Simplify installation with Go instrumentation tool'
+tags:
+  - Agents
+  - Go agent
+  - Installation
+metaDescription: How to use a script to get code suggestions to instrument your Golang apps and services.
+freshnessValidatedDate: 2024-08-20 
+---
+<Callout title="preview">
+  We're still working on this feature, but we'd love for you to try it out!
+
+  This feature is currently provided as part of a preview program pursuant to our [pre-release policies](/docs/licenses/license-information/referenced-policies/new-relic-pre-release-policy).
+</Callout>
+
+Go is a compiled language with an opaque runtime, making it unable to support automatic instrumentation like other languages. For this reason, the New Relic Go agent is designed as an SDK. Since the Go agent is an SDK, it requires more manual work to set up than agents for languages that support automatic instrumentation.
+
+In an effort to reduce that manual effort, the Go agent team created an easy instrumentation tool for Go that is currently in preview. This tool will do most of the work for you by suggesting changes to your source code that instrument your application with the New Relic Go agent.
+
+To get started, check out this [four-minute video](https://asciinema.org/a/r0Il7o2eMiZaLKHIlew3IL2nx), or skip down to [How it works](#how-it-works):
+
+## Preview notice [#preview-notice]
+
+This feature is currently provided as part of a product preview and is subject to our New Relic Experimental policies. Recommended code changes are suggestions only and should be subject to human review for accuracy, applicability, and appropriateness for your environment. This feature should only be used in non-critical, non-production environments that do not contain sensitive data.
+
+This project, its code, and the UX are under heavy development, so you should expect it to change. Please take this into consideration when participating in this preview. If you encounter any issues, please report them using [Github issues](https://github.com/newrelic/go-agent-assisted-instrumentation) and fill out as much of the issue template as you can so we can improve this tool.
+
+## How it works [#how-it-works]
+
+This tool will not interfere with your application's operation, and it doesn't make any changes to your code directly. Here's what happens:
+
+* It analyzes your source code and identifies opportunities to instrument it.
```

Comment:
"opportunities" is odd... Does it identify methods to instrument or simply places in the code where you can add the instrumentation code?

Reply:
"It" feels weird too. Can we get more specific? 

I feel like there are really two steps to this: 
1. You run the tool. It analyzes the code and suggests changes to your code.
2. You review the changes in the .diff file and decide which changes to add to your source code.

Reply:
@paperclypse I went with this revision:

* It analyzes your code and suggests changes that allow the Go agent to capture telemetry data.      
* You review the changes in the `.diff` file and decide which changes to add to your source code.